### PR TITLE
Add JSONAPI data member validation

### PR DESCRIPTION
**Pull request by ThomasKoppensteiner**

### CODE REVIEW

**Review by ghsmaniotto — DISMISSED** [review]

**Review by diendorfersarah — DISMISSED** [review]

File: lib/request_handler/json_api_document_parser.rb [hunk 1]
```diff
@@ -27,9 +29,22 @@ def flattened_document
...
       end
+      validate_json_api_data_members(resource)
       flatten_resource!(resource)
     end
 
+    def validate_json_api_data_members(data)
+      data.keys.each do |k|
+        next if VALID_DATA_MEMBERS.include?(k)
+
+        raise BodyParamsError, [{ code: 'INVALID_JSON_API',
```
Comment (Goltergaul) [inline lib/request_handler/json_api_document_parser.rb:40]: afaik the requesthandler does per default not verify if additional fields are passed in somewhere. Even though i like checking for extra things very much, I'm not sure if introducing this now is maybe not breaking something somewhere

**Comment by Goltergaul** [comment]
why are you adding this validation step? What's the context?

File: lib/request_handler/json_api_document_parser.rb (hunk 1 above)
Comment (ThomasKoppensteiner) [reply lib/request_handler/json_api_document_parser.rb:40]: See PM

**Review by diendorfersarah — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## ## [3.0.0] - 2023-02-07
+### Added
+- JSONAPI data member validation
```
Comment (Goltergaul) [inline CHANGELOG.md:9]: - Only "links" and "meta" are allowed besides id, type and attributes in "data". If anything else is present, an error is now raised.

makes it clearer what this is about and what can now potentially go wrong

**Pull request merged**